File: src/spirite/pc/Spirite.java
// Rory Burks

package spirite.pc;

import spirite.base.brains.MasterControl;
import spirite.hybrid.MDebug;
import spirite.hybrid.MDebug.ErrorType;
import spirite.pc.ui.omni.FrameManager;

import javax.swing.*;



/**
 * Entry Point
 * 
 * TODO: Make all Observers weak-referenced.
 */
public class Spirite{
    
    public static MasterControl master;

    public static void main(String[] args) {
       	try {
            master = new MasterControl();

            setLookAndFeel();

            // So that it's run on the UI thread.
            SwingUtilities.invokeLater( new Runnable() {
                public void run() { createUI();}
            });
        } catch( Exception e) {
        	MDebug.handleError( ErrorType.FATAL, e, "Core:" + e.getMessage());
        }
    }

    public static void setLookAndFeel() {
        try {
            UIManager.setLookAndFeel( UIManager.getCrossPlatformLookAndFeelClassName());
    		//UIManager.getLookAndFeelDefaults().put( "background" , new ColorUIResource(new Color(152, 166, 173) ));
    		//UIManager.getLookAndFeelDefaults().put( "Panel.background" , new ColorUIResource(new Color(152, 166, 173) ));
        }
        catch (Exception e) {
        	MDebug.handleError( ErrorType.RESOURCE, e, "Invalid Look and Feel.");
        }
    }

    public static void createUI() {
    	FrameManager frame_manager = master.getFrameManager();
    	frame_manager.packMainFrame();
    }
}

File: src/main/kotlin/spirite/pc/Spirite.kt
package spirite.pc

import rbJvm.vectrix.SetupVectrixForJvm
import spirite.base.brains.MasterControl
import spirite.base.imageData.mediums.MediumType.DYNAMIC
import spirite.core.hybrid.DebugProvider
import spirite.core.hybrid.IDebug.ErrorType.FATAL
import spirite.pc.di.SetupHybrid
import spirite.sguiHybrid.EngineLaunchpoint
import javax.swing.SwingUtilities
import javax.swing.UIManager


fun main( args: Array<String>) {
    println()
    Spirite().run()
}

class Spirite {
    lateinit var master: MasterControl

    var ready : Boolean  = false ; private set

    fun run() {
        try {
            SetupVectrixForJvm()
            setupSwGuiStuff()
            SetupHybrid()

            UIManager.setLookAndFeel( UIManager.getCrossPlatformLookAndFeelClassName())
            SwingUtilities.invokeAndWait {
                EngineLaunchpoint.gle
                master = MasterControl()

                master.frameManager.initUi()}


            SwingUtilities.invokeLater {
                val ws1 = master.createWorkspace(640,480)
                ws1.groupTree.addNewSimpleLayer(null, "Background", DYNAMIC)
                master.workspaceSet.addWorkspace(ws1)
                ws1.finishBuilding()
                ready = true
            }
        }catch (e : Exception) {
            e.printStackTrace()
            DebugProvider.debug.handleError(FATAL, e.message ?: "Root-level exception caught.", e)
        }
    }
}
